examples/client-side-validation/app/routes/example.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  ActionFunction,
+  json,
+  LinksFunction,
+  useActionData
+} from "remix";
+import stylesUrl from "../styles/index.css";
+
+export const links: LinksFunction = () => {
+  return [
+    {
+      rel: "stylesheet",
+      href: stylesUrl
+    }
+  ];
+};
+
+export const action: ActionFunction = async ({ request }) => {
+  const form = await request.formData();
+  const message = `Successfully submitted data:
+      - Required text: ${form.get("required-text")}
+      - Required checkbox: ${form.get("required-checkbox")}
+      - Text with regex: ${form.get("text-with-regex")} 
+      - Number with min max: ${form.get("number-with-min-max")}
+      - Text with minlength maxlength: ${form.get("text-with-minlength-maxlength")}
+      - Date with min max: ${form.get("date-with-min-max")}
+  `;
+  return json({ message }, { status: 200 });
+};
+
+export default function Example() {
+  const actionData = useActionData();
+  const date = new Date();
```

Comment:
This will have SSR hydration issues since the server and client can be in different timezones. It would probably be better if these values came from the loader to avoid that.